### DISCUSSION TITLE
日历中添加自选显示的节假日

### DISCUSSION THREAD

**Discussion by ShuiHuo** [linuxdeepin/developer-center#5782] [Features Request & Ideas | 特性请求 & 头脑风暴]
需要录入更多全国和地方节假日、纪念日，然后用户可以自己选择是否显示哪里、哪种、哪个日子。比如我想显示全国和云南的日子，那我就勾选这两个地方，又比如我不需要愚人节，那我可以取消显示这个节日。

**Comment by BLumia** [comment]
cc @meiyixiang @myml

**Comment by myml** [comment]
新版本日历 5.11.0 添加了ics导入功能，你可以找到地方节假日的ics文件导入到日历中，比如 https://github.com/lanceliao/china-holiday-calender

**Reply by ShuiHuo** [reply]
在 5.11.0 里面没找到导入功能，但我猜这功能和我的需求不符。

　　一般导入日程只会显示为日程的样式，日程有效时间较短，没有农历（或农历格式不统一）等。

**Reply by myml** [reply]
5.11.0 在设置->日程类型管理旁边添加了日程的导入和导出，可以寻找别人创建的地方农历ics 或者 自己创建后分享出来。
没明白你说的有效时间较短的意思，日历创建日程是支持按年重复的，导出ics文件也支持。所以如果将日程设置成按年重复，是无限有效时间的。
日历导出ics时通过扩展字段实现了农历支持，这个可能和其他日历程序存在兼容问题，但ics文件改起来也不难，真正困难的是地方节假日搜集。。。

**Reply by ShuiHuo** [reply]
找到导入导出了。

　　日程有效期较短是指：部分节日不是指定几月几日，而是多少年几月第几个什么，又像我另一个讨论里面说的某个节气之前，这种情况还不能直接重复，好像也能改改实现支持是吧。支持农历的日历日程软件大多使用了不同的扩展字段，还有个 RFC 7529，感觉设计得都不咋样（RFC 7529 比较有意思的就是农历的年份了），没有真正往易读写和可扩展方向去做。

　　节假日收集……在官网做成长期任务收集可能能稍微解决一点……

**Reply by myml** [reply]
日历之后是计划支持Exchange同步和WebCal订阅以及遗留的BUG修复，目前来看这个需求的有些太小众了，我这边可能不会排时间去做。
